From 2c81e75a4471cb42561f0a6bb35362198f986bfa Mon Sep 17 00:00:00 2001
From: Moritz Fischer <moritz.fischer@ettus.com>
Date: Wed, 10 May 2017 22:24:04 -0700
Subject: [PATCH 07/14] ni: zynq: Mender.io support for NI Ettus Research
 Sulfur Rev2 SDR

This involves adding bootcount in environment, as well as modifying
the default bootconfiguration to run ${mender_setup} first.

Signed-off-by: Moritz Fischer <moritz.fischer@ettus.com>
---
 configs/ni_sulfur_rev2_defconfig |  2 +-
 include/configs/ni_sulfur_rev2.h | 11 ++++-------
 2 files changed, 5 insertions(+), 8 deletions(-)

diff --git a/configs/ni_sulfur_rev2_defconfig b/configs/ni_sulfur_rev2_defconfig
index ee5af861a7e..5e15140f886 100644
--- a/configs/ni_sulfur_rev2_defconfig
+++ b/configs/ni_sulfur_rev2_defconfig
@@ -3,7 +3,7 @@ CONFIG_SYS_VENDOR="ni"
 CONFIG_SYS_CONFIG_NAME="ni_sulfur_rev2"
 CONFIG_ARCH_ZYNQ=y
 CONFIG_DEFAULT_DEVICE_TREE="zynq-ni-sulfur-rev2"
-CONFIG_SYS_EXTRA_OPTIONS="ENV_IS_IN_MMC"
+CONFIG_SYS_EXTRA_OPTIONS="BOOTCOUNT_LIMIT,BOOTCOUNT_ENV,ENV_IS_IN_MMC"
 CONFIG_FIT=y
 # CONFIG_DISPLAY_CPUINFO is not set
 CONFIG_SPL=y
diff --git a/include/configs/ni_sulfur_rev2.h b/include/configs/ni_sulfur_rev2.h
index 1165104829f..3e053bc05d0 100644
--- a/include/configs/ni_sulfur_rev2.h
+++ b/include/configs/ni_sulfur_rev2.h
@@ -14,13 +14,10 @@
 #include "zynq-common.h"
 #undef CONFIG_ENV_IS_NOWHERE
 
-#define CONFIG_SYS_MMC_ENV_DEV 0
-#define CONFIG_ENV_SECT_SIZE		CONFIG_ENV_SIZE
-#define CONFIG_ENV_OFFSET		0xE0000
-
 #undef CONFIG_EXTRA_ENV_SETTINGS
 #define CONFIG_EXTRA_ENV_SETTINGS	\
-	"fit_image=fit.itb\0"		\
+	"fit_image=boot/fitImage\0"	\
+	"bootargs=root=${mender_kernel_root} rw rootwait\0" \
 	"load_addr=0x2000000\0"		\
 	"fit_size=0x800000\0"		\
 	"fdt_high=0x20000000\0"		\
@@ -49,8 +46,8 @@
 				"run uenvcmd; " \
 			"fi; " \
 		"fi; \0" \
-	"sdboot=echo Copying FIT from SD to RAM... && " \
-		"load mmc 0 ${load_addr} ${fit_image} && " \
+	"sdboot=run mender_setup; echo Copying FIT from SD to RAM... && " \
+		"ext4load ${mender_uboot_root} ${load_addr} ${fit_image} && " \
 		"bootm ${load_addr}\0" \
 	"jtagboot=echo TFTPing FIT to RAM... && " \
 		"tftpboot ${load_addr} ${fit_image} && " \
-- 
2.14.2

